components/BackdropMap.tsx

```diff
@@ -12,25 +12,19 @@ export default function BackdropMap() {
     const [zoom, setZoom] = useState(9)
 
   useEffect(() => {
-    if (map.current) return;
-
     map.current = new mapboxgl.Map({
-      container: mapContainer.current || "",
+      container: mapContainer.current!,
       style: "mapbox://styles/mapbox/streets-v12",
       center: [lng, lat],
       zoom: zoom
     })
-  })
-
-  useEffect(() => {
-    if (!map.current) return;
 
     map.current.on('move', () => {
       setLng(Number(map.current?.getCenter().lng.toFixed(4)));
       setLat(Number(map.current?.getCenter().lat.toFixed(4)));
       setZoom(Number(map.current?.getZoom().toFixed(2)));
     });
-  });
+  }, [])
```

Comment:
`zoom` might need to be in this array?

Reply:
My reasoning for omitting it is that I only want that hook to run once to instantiate the map and assign the event listener there on lines 22-26 allowing the user to move the map around. Since the user has no way to change the`zoom` state before that first render occurs, and after the first render they'll be changing it using the event listener added on line 22, I thought it'd probably be safe to just leave it out.

Reply:
Cool!